tuyaha/tuyaapi.py

```diff
@@ -1,19 +1,26 @@
-import json
 import logging
 import time
 
 import requests
+from datetime import datetime
 from requests.exceptions import ConnectionError as RequestsConnectionError
 from requests.exceptions import HTTPError as RequestsHTTPError
+from threading import Lock
 
 from tuyaha.devices.factory import get_tuya_device
 
 TUYACLOUDURL = "https://px1.tuya{}.com"
 DEFAULTREGION = "us"
 
+MIN_DISCOVERY_INTERVAL = 60.0
+DEF_DISCOVERY_INTERVAL = 305.0
```

Comment:
305 is 5 minutes + 5 seconds.
This is because Tuya API now allow  discovery once every 300 seconds (reported as error by API itself if called with higher freq.). Because sometimes waiting exactly 300 seconds is not enought, I add 5 second for security.

Reply:
Anyway I review this values

Reply:
Just change this value to what you think will be correct and add a comment on why you decided to use this magic number.

Reply:
Write something that will help next developer that will look into this code. Something that you wrote in your comment https://github.com/PaulAnnekov/tuyaha/pull/39#discussion_r491719432 but shorter, like `Tuya API limits Discovery requests to one per 300 seconds and Query requests to one per 60 seconds. We also adding +5 seconds to be sure we will not get an error`